cfme/tests/configure/test_access_control.py

```diff
@@ -42,6 +44,39 @@
 SETTINGS = VersionPicker({LOWEST: "Settings", "5.11": "User Settings"})
 
 
+def create_custom_user(appliance, provider):
+    """create a role, group, user with compute only role product features"""
+    # Creating Role
+    role = appliance.collections.roles.instantiate(name="EvmRole-user")
+    custom_role = role.copy(name=fauxfactory.gen_alphanumeric(30, start="EvmRole-user_copy_"))
+    with update(custom_role):
+        custom_role.product_features = [
+            (["Everything"], True),
+            (["Everything"], False),
+            (["Everything", "Compute"], True),
+        ]
+
+    # Creating Group
+    group = appliance.collections.groups.create(
+        description=fauxfactory.gen_alphanumeric(30, "group_description_"),
+        role=custom_role.name,
+        tenant="My Company",
+        host_cluster=([provider.data["name"]], True),
+    )
+
+    # Creating User
+    user = appliance.collections.users.create(
+        name=fauxfactory.gen_alphanumeric(start="user_").lower(),
+        credential=Credential(
+            principal=fauxfactory.gen_alphanumeric(start="uid"),
+            secret=fauxfactory.gen_alphanumeric(start="pwd"),
+        ),
+        email=fauxfactory.gen_email(),
+        groups=[group],
+    )
+    return custom_role, group, user
```

Comment:
Alternatively, you can use REST to create role, group and user to make this thing faster and you will also be free from taking care of teardown.

In case you decide to go via REST, you will need `ems_container` and `container_group` features which you can pass to `cfme.rest.gen_data.py::roles()` to create your desired role. You can use `cfme.rest.gen_data.py::users()` and `cfme.rest.gen_data.py::groups()` to create user and group. You can see `cfme/tests/configure/test_rbac_landing_page.py::setup_user()` fixture to see how to implement this thing.

Reply:
After discussion on gchat keeping code as it is.